### DISCUSSION TITLE
GraalVM Hint Processor nice with JBang for GraalVM native-image

### DISCUSSION THREAD

**Discussion by linux-china** [jbangdev/jbang#1330] [Show and tell]
GraalVM Hint Processor helps generate GraalVM hints for building [native-image](https://www.graalvm.org/reference-manual/native-image/) applications.   https://github.com/GoodforGod/graalvm-hint 

It works well with JBang for GraalVM native application, and make GraalVM native app easy without configuration json files involved. 

```java
///usr/bin/env jbang "$0" "$@" ; exit $?
//JAVA 17
//DEPS io.goodforgod:graalvm-hint-processor:0.17.1
//DEPS io.goodforgod:graalvm-hint-annotations:0.17.1
//DEPS com.fasterxml.jackson.core:jackson-databind:2.13.2.2

import com.fasterxml.jackson.databind.ObjectMapper;
import io.goodforgod.graalvm.hint.annotation.ReflectionHint;

public class Hello {
    public static void main(String... args) throws Exception {
        ObjectMapper objectMapper = new ObjectMapper();
        User user = new User(1, "linux_china");
        String jsonText = objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(user);
        final User user2 = objectMapper.readValue(jsonText, User.class);
        System.out.println(user2.getNick());
    }
}

@ReflectionHint
class User {
    private Integer id;
    private String nick;

    public User() {
    }

    public User(Integer id, String nick) {
        this.id = id;
        this.nick = nick;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getNick() {
        return nick;
    }

    public void setNick(String nick) {
        this.nick = nick;
    }
}

```

**Comment by quintesse** [comment]
Cool stuff!

**Comment by maxandersen** [comment]
Cool - looks like quarkus but with different annotations ?

**Reply by quintesse** [reply]
I have the feeling it's something much simpler. Where instead of having to deal with Graal configuration files you use annotations to tell it how to treat the code. There's no "core" that you need to add, nor does it seem to add anything to what GraalVM supports by default. And on top of that you need to know what you're doing :slightly_smiling_face:

**Reply by maxandersen** [reply]
Sure but that's what quarkus does (or one thing it does). Let you use annotations instead of having to deal with reams of json

**Reply by quintesse** [reply]
Ok, but can you use Quarkus to do _only_ that? (And not have to depend on Quarkus for anything in your code)
